main.go

```diff
@@ -42,6 +43,14 @@ func hmacValidation() bool {
 	return len(val) > 0 && (val == "1" || val == "true")
 }
 
+func removeNewLine(secret []byte) []byte {
```

Comment:
This looks perfect - spot on. I would call this function something more constructive like `getFirstLine` or similar. 

Other than that test looks good to me too.